dockers/docker-dhcp-relay/cli-plugin-tests/test_show_dhcp4relay_counters.py

```diff
@@ -0,0 +1,387 @@
+    ("Ethernet1", "Vlan1000", "Downlink"),
+    ("Ethernet4", "Vlan1000", "Uplink"),
+    ("Ethernet5", "Vlan3000", "Unknown"),
+    ("eth0", "Vlan3000", "MGMT")
+])
+def test_get_interfaces_type(interface_name, vlan_interface, expected_result):
+    vlan_members = {
+        "Vlan1000": set(["Ethernet1", "Ethernet2"]),
+        "Vlan2000": set(["Ethernet3"])
+    }
+    mgmt_intfs = ["eth0"]
+    result = show_dhcp_relay.get_interfaces_type(interface_name, mgmt_intfs, vlan_interface, vlan_members)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize("interfaces, dirs_count, vlan_interface", [
+    (["Ethernet1", "Ethernet2", "Ethernet3", "Ethernet4"], {"RX": "1", "TX": "5"}, "Vlan1000"),
+    (["eth0", "Ethernet3", "Ethernet4", "Ethernet5"], {"RX": "3 ", "TX": "5"}, "Vlan3000"),
+])
+def test_append_interfaces_count_with_type_specified(interfaces, dirs_count, vlan_interface):
+    def mock_get_count_from_db(db, key, current_dir, current_type):
+        return dirs_count[current_dir]
+
+    vlan_members = {
+        "Vlan1000": set(["Ethernet1", "Ethernet2"]),
+        "Vlan2000": set(["Ethernet3"])
+    }
+    mgmt_intfs = ["eth0"]
+    with patch.object(show_dhcp_relay, "get_count_from_db", side_effect=mock_get_count_from_db):
+        dummy_summary = "dummy"
+        data = {
+            dummy_summary: [],
+            "Interface Type": []
+        }
+        for dir in dirs_count.keys():
+            data[dir] = []
+        expected_data = copy.deepcopy(data)
+        mock_db = MagicMock()
+        mock_db.keys.return_value = ["DHCPV4_COUNTER_TABLE:" + vlan_interface + ":" + intf for intf in interfaces]
+        show_dhcp_relay.append_interfaces_count_with_type_specified(data, mock_db, list(dirs_count.keys()),
+                                                                    vlan_interface, "", dummy_summary, vlan_members,
+                                                                    mgmt_intfs)
+        expected_data[dummy_summary] = interfaces
+        for intf in interfaces:
+            if intf in mgmt_intfs:
+                expected_data["Interface Type"].append("MGMT")
+            elif vlan_interface not in vlan_members:
+                expected_data["Interface Type"].append("Unknown")
+            elif intf in vlan_members[vlan_interface]:
+                expected_data["Interface Type"].append("Downlink")
+            else:
+                expected_data["Interface Type"].append("Uplink")
+            for dir in dirs_count.keys():
+                expected_data[dir].append(dirs_count[dir])
+        assert expected_data == data
+
+
+def test_generate_output_with_type_specified():
+    expected_output = """\
++--------------------+------------------+------+------+
+|   Vlan1000 (Offer) |   Interface Type |   RX |   TX |
++====================+==================+======+======+
+|           Vlan1000 |             VLAN |    5 |    5 |
++--------------------+------------------+------+------+
```

Comment:
The output is not aligned with HLD, can you remove those additional lines `+--------------------+------------------+------+------+` ?
```
| Ethernet1     | Downlink  | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   | 0   |
| Ethernet2     | Downlink  | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   | 0   |
| Ethernet46    | Uplink    | 0   | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   |
| Ethernet47    | Uplink    | 0   | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   |
| PortChannel1  | Uplink    | 0   | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   |
| PortChannel2  | Uplink    | 0   | 0   | 2   | 0   | 2   | 0   | 0   | 0   | 0   |
```

Reply:
Fixed